src/cli/add.rs

```diff
@@ -111,18 +124,137 @@ pub async fn execute(args: Args) -> miette::Result<()> {
         .collect::<Vec<Platform>>();
     project.add_platforms(platforms_to_add.iter())?;
 
-    add_specs_to_project(
-        &mut project,
-        args.specs,
-        spec_type,
-        args.no_install,
-        args.no_lockfile_update,
-        spec_platforms,
-    )
-    .await
+    match spec_type {
+        SpecType::Host | SpecType::Build | SpecType::Run => {
+            let specs = args
+                .specs
+                .into_iter()
+                .map(|s| MatchSpec::from_str(&s))
+                .collect::<Result<Vec<_>, _>>()
+                .into_diagnostic()?;
+            add_conda_specs_to_project(
+                &mut project,
+                specs,
+                spec_type,
+                args.no_install,
+                args.no_lockfile_update,
+                spec_platforms,
+            )
+            .await
+        }
+        SpecType::Pypi => {
+            // Parse specs as pep508_rs requirements
+            let pep508_requirements = args
+                .specs
+                .into_iter()
+                .map(|input| pep508_rs::Requirement::from_str(input.as_ref()).into_diagnostic())
+                .collect::<miette::Result<Vec<_>>>()?;
+
+            // Move those requirements into our custom PyPiRequirement
+            let specs_result: miette::Result<Vec<(rip::types::PackageName, PyPiRequirement)>> =
+                pep508_requirements
+                    .into_iter()
+                    .map(|req| {
+                        let name = rip::types::PackageName::from_str(req.name.as_str())?;
+                        let requirement = PyPiRequirement::from(req);
+                        Ok((name, requirement))
+                    })
+                    .collect();
```

Comment:
Why not just add the question mark here? Instead of mapping the result? Then the code below no longer needs the `map`.

Reply:
Changed it